### PR TITLE
Rewrite test_users_groups to pytest style

### LINKED ISSUE #3334: Refactor `test_users_groups` test module to use pytest fixtures
The `test_users_groups` module[0] needs to be refactored to use pytest fixtures. 

[0] https://github.com/pulp/pulpcore/blob/main/pulpcore/tests/functional/api/test_users_groups.py

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #3334

### CODE REVIEW

**Review by dkliban — APPROVED** [review]

File: pulpcore/tests/functional/api/test_users_groups.py [hunk 1]
```diff
@@ -1,99 +1,60 @@
...
+@pytest.mark.parallel
+def test_filter_groups(groups_api_client, gen_object_with_cleanup):
+    """Test that groups can be filterd."""
+
+    prefix = str(uuid.uuid4())
+    groups = groups_api_client.list(name__contains=f"{prefix}_")
+    assert len(groups.results) == 0
+
+    gen_object_with_cleanup(groups_api_client, {"name": f"{prefix}_newbees"}).pulp_href
+    gen_object_with_cleanup(groups_api_client, {"name": f"{prefix}_admims"}).pulp_href
```
Comment (ipanova) [inline pulpcore/tests/functional/api/test_users_groups.py:53]: s/admims/admins

File: pulpcore/tests/functional/api/test_users_groups.py [hunk 2]
```diff
@@ -1,99 +1,60 @@
...
+def test_crud_groups(groups_api_client, gen_object_with_cleanup):
+    """Test that a group can be crud."""
+
+    prefix = str(uuid.uuid4())
+    groups = groups_api_client.list()
+    assert f"{prefix}_newbees" not in [group.name for group in groups.results]
+    group_href = gen_object_with_cleanup(groups_api_client, {"name": f"{prefix}_newbees"}).pulp_href
+    groups = groups_api_client.list()
+    assert f"{prefix}_newbees" in [group.name for group in groups.results]
+    groups_api_client.delete(group_href)
```
Comment (ipanova) [inline pulpcore/tests/functional/api/test_users_groups.py:39]: any reason call delete() if we have used `gen_object_with_cleanup`?

File: pulpcore/tests/functional/api/test_users_groups.py [hunk 3]
```diff
@@ -1,99 +1,60 @@
...
+    users = users_api_client.list(username__contains=f"{prefix}_")
+    assert len(users.results) == 2
+    users = users_api_client.list(username__contains=f"{prefix}_new")
+    assert len(users.results) == 1
+    users = users_api_client.list(username=f"{prefix}_newbee")
+    assert len(users.results) == 1
+
+
+@pytest.mark.parallel
+def test_crud_groups(groups_api_client, gen_object_with_cleanup):
```
Comment (ipanova) [inline pulpcore/tests/functional/api/test_users_groups.py:30]: this test does not exactly test of the CRUD operations ;)

File: pulpcore/tests/functional/api/test_users_groups.py [hunk 4]
```diff
@@ -1,99 +1,60 @@
...
+
+
+@pytest.mark.parallel
+def test_filter_users(users_api_client, gen_user):
+    """Test that users can be filterd."""
+    prefix = str(uuid.uuid4())
+    users = users_api_client.list(username__contains=f"{prefix}_")
+    assert len(users.results) == 0
+
+    gen_user(username=f"{prefix}_newbee")
```
[the remark this answers is not in the document]
Comment (ipanova) [reply pulpcore/tests/functional/api/test_users_groups.py:18]: yes, https://github.com/pulp/pulpcore/blob/main/pulpcore/tests/functional/__init__.py#L316

File: pulpcore/tests/functional/api/test_users_groups.py (hunk 2 above)
Comment (mdellweg) [reply pulpcore/tests/functional/api/test_users_groups.py:39]: Yes, so we can list it again and see it's gone. The `gen_object_with_cleanup` is used so we clean even if a test failed.

File: pulpcore/tests/functional/api/test_users_groups.py (hunk 3 above)
Comment (mdellweg) [reply pulpcore/tests/functional/api/test_users_groups.py:30]: Changing to CRD.

File: pulpcore/tests/functional/api/test_users_groups.py (hunk 2 above)
Comment (ipanova) [reply pulpcore/tests/functional/api/test_users_groups.py:39]: gotcha that makes sense

**Review by ipanova — APPROVED** [review]

**Pull request merged**